dist/categorical.go

```diff
@@ -0,0 +1,177 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dist
+
+import (
+	"math"
+	"math/rand"
+)
+
+// Categorical is an extension of the Bernouilli distribution where x can take
+// a certain number of possible values. Categorical must be initialized with
```

Comment:
Thanks for reminding me of this. I was wondering why the 0-basedness was mentioned. Isn't everything we do 0-based? A description like @vladimir-ch has would be more informative.